### PR TITLE
fuzz: update Generex

### PR DESCRIPTION
**Pull request by thc202**
Update Generex to latest version, 1.0.2.
Add a custom iterator to calculate the number of payloads that can be
generated from a regex up to a given limit, update payload generator
accordingly.

Fix zaproxy/zaproxy#4751.

### CODE REVIEW

File: addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/payloads/generator/StateStringCounter.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
```
[the remark this answers is not in the document]
Comment (thc202) [reply addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/payloads/generator/StateStringCounter.java:6]: It is.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/payloads/generator/StateStringCounter.java:6]: OMG, I totally saw 2011.....I'm sooooo sorry. 

/me wanders off for more coffee 😢

**Review by kingthorin — APPROVED** [review]

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
Will fix the conflict.

**Comment by thc202** [comment]
Conflict addressed.

**Review by kingthorin — APPROVED** [review]

File: addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/payloads/generator/StateStringCounter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (thc202) [reply addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/payloads/generator/StateStringCounter.java:6]: No worries :)

**Pull request merged**